### PR TITLE
[CPAOT] Generating code for hardware intrinsic (#26772)

### PR DESCRIPTION
**Pull request by cshung**

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
Is there something in master that prevents us from doing a full integration? David did a full integration a week ago. There's not much stuff in master that is not in single-exe.

[Cherry picking commits](https://devblogs.microsoft.com/oldnewthing/?p=98215) can cause merging trouble down the line. Especially because we expect to make changes in these lines soon.

File: src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs [hunk 1]
```diff
@@ -1154,6 +1163,109 @@ private static bool IsDynamicStatics(TypeDesc type)
...
+                        }
+                    }
+                    else if (namespaceName == "System")
+                    {
+                        if ((className == "Math") || (className == "MathF"))
+                        {
+                            // These are normally handled via the SSE4.1 instructions ROUNDSS/ROUNDSD.
+                            // However, we don't know the ISAs the target machine supports so we should
+                            // fallback to the method call implementation instead.
+                            fTreatAsRegularMethodCall = (methodName == "Round");
```
Comment (EgorBo) [inline src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs]: Same for `Floor` and `Ceiling` ?

Comment (MichalStrehovsky) [reply src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs]: Good point!

Those are handled at a different spot though. @cshung Looks like you missed porting the `getIntrinsicID` part of #25365.

Comment (cshung) [reply src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs]: Thanks for the feedback - yes it is indeed missed.

**Comment by cshung** [comment]
> [Cherry picking commits](https://devblogs.microsoft.com/oldnewthing/?p=98215) can cause merging trouble down the line. Especially because we expect to make changes in these lines soon.

I was doing this `cherry-pick` PR just because Simon asked me to. To be honest I was thinking about `rebase` which is probably not what we wanted either. I will do a separate `merge` PR instead.

**Pull request closed without merging**